### PR TITLE
Enable strict classpath scanning optionally

### LINKED ISSUE #51: Throw exceptions when classes cannot be parsed
As followup to https://github.com/eclipse/sisu.inject/issues/97 [PlexusAnnotatedBeanModule](https://github.com/eclipse/sisu.plexus/blob/5e8587eac8c4c079566dcd0ccb0d5dba40cb4169/org.eclipse.sisu.plexus/src/main/java/org/eclipse/sisu/plexus/PlexusAnnotatedBeanModule.java#L76) should leverage strict scanning.

### PR DESCRIPTION
**Pull request by kwin**
This closes #51

### CODE REVIEW

File: org.eclipse.sisu.plexus/src/main/java/org/codehaus/plexus/ContainerConfiguration.java [hunk 1]
```diff
@@ -62,4 +62,9 @@ public interface ContainerConfiguration
     ContainerConfiguration setJSR250Lifecycle( boolean on );
 
     boolean getJSR250Lifecycle();
+
+    ContainerConfiguration setStrictClassPathScanning( boolean strictScanning );
```
Comment (kwin) [inline org.eclipse.sisu.plexus/src/main/java/org/codehaus/plexus/ContainerConfiguration.java:66]: It seems this interface was anyhow much modified with Sisu Plexus in comparison to the last known version at https://github.com/codehaus-plexus/plexus-containers/blob/3b0a8d5fd086e3f74ccdeecaac9cd63e44047e91/plexus-container-default/src/main/java/org/codehaus/plexus/ContainerConfiguration.java#L22. So adding two more methods doesn't do any harm here.

Comment (cstamas) [reply org.eclipse.sisu.plexus/src/main/java/org/codehaus/plexus/ContainerConfiguration.java:66]: Exactly, config was had to be changed (scanning, auto wire, etc)

**Review by cstamas — APPROVED** [review]

**Comment by cstamas** [comment]
My guess this depends on sisu.inject master...

**Comment by kwin** [comment]
> My guess this depends on sisu.inject master...

Yes, one more reason for a common repo/build…

**Comment by cstamas** [comment]
merge master, added updatePolicy=always to snapshot repo (which makes sense)

**Pull request merged**